### DISCUSSION TITLE
updateTrigger getRadius

### DISCUSSION THREAD

**Discussion by peter81git** [visgl/deck.gl#5488] [Q&A] [answered]
I'm trying to update radius for my GeoJsonLayer like so

``` js
myGeoJsonLayer = new MapboxLayer({
              id: 'my-gjson',
              type: GeoJsonLayer,
              data:out,
              pointRadiusScale:1.0,
              getRadius: 1700*(1/this.zoomRadius),
              updateTriggers: {
                getRadius:this.zoomRadius

              },
              getFillColor: f=> this.fillColorLayer(f.properties),
              getLineColor: [255, 255, 255],
              extruded:true,
              pickable: true,
              onClick: (info, event) => this.onclickPop(info,event),
            });
```

But even if the zoomRadius change there is no visual change/update. I've checked the variable and it is updating when zooming.
Zoom: 11.601
Zoom: 11.671
Zoom: 11.745
Zoom: 11.819
Zoom: 11.855
Zoom: 11.983
Zoom: 12.021
Zoom: 12.119
Zoom: 12.122
Zoom: 12.183
Zoom: 12.213

Is anything else that I should configure on top?

Versions are:
@deck.gl/core                          8.3.13
@deck.gl/layers                        8.3.13
@deck.gl/mapbox                    8.3.13

Thank you.
Peter

**Comment by Pessimistress** [comment, marked as answer]
What is your `this.zoomRadius`?

`zoom` is logarithmic, i.e. the same meter radius looks twice as big from z to z+1.

You can also consider using [pointRadiusUnits: 'pixels'](https://deck.gl/docs/api-reference/layers/geojson-layer#pointradiusunits).

Update trigger is not needed unless `getRadius` is a function.

**Reply by peter81git** [reply]
The
```js
 this.zoomRadius
``` 
is a variable that is updated whenever on 'zoom' event is 'fired'.

**Reply by Pessimistress** [reply]
I can't tell you how to fix it if you don't share how you are calculating this variable. Most likely your change to `getRadius` is too small to make a visual difference. Please refer to my comment above.

**Reply by peter81git** [reply]
Currently I'm updating that variable like so:
```js 
map.on('zoom', ()=> {
           this.zoomRadius = map.getZoom().toFixed(3);
  });
```
giving for example Zoom: 13.102;  Zoom: 13.140; Zoom: 13.167.

I've tested also like bellow, that uses map.getZoom() directly:
```js getRadius: f=> 1700*(1/map.getZoom()),
              updateTriggers: {
                getRadius:map.getZoom()

              },
```
and even like that radius only changes if for example I use a button click directly to: 
```js
myGeoJsonLayer.setProps({
  updateTriggers: {
                getRadius:map.getZoom()

     }
});
```

Hopefully this is a bit more clear. :)

**Reply by Pessimistress** [reply]
As I said, zoom is logarithmic. Any visible change would involve  something like

```js
getRadius: R_IN_METERS / (2 ** zoom)
```

Which is equivalent to

```js
getRadius: R_IN_PIXELS,
pointRadiusUnits: 'pixels'
```

**Reply by peter81git** [reply]
With getRadius in pixels units worked ok.
In the future I'll try with meters.
Just one question. I'm using deck+mapbox that is why I'm using map.getZoom(). The `zoom` you are specifying is directly available in deck?
